### PR TITLE
Make builds minimal

### PR DESCRIPTION
**Pull request by stevecrozz**
This is in reference to #1711 where I suggest we package minimal builds that only include the essentials. This change saves me about 2MiB on disk which sounds like nothing, but on my alpine linux Docker images its actually significant. Since we started discussing already, its probably best to talk about the general idea of removing non-essential files from the build in #1711, but here's a concrete example of what I'm thinking.

### CODE REVIEW

**Comment by tenderlove** [comment]
This seems alright to me.  Agree with @flavorjones that it's sometimes helpful to have the tests, but I also don't feel strongly about it (especially since we tag the release version in git anyway).

**Comment by flavorjones** [comment]
OK - I agree in principle with minimizing the gem; however I want to make sure I have automated testing of the generated gem to ensure that we're not omitting necessary files. This is at the top of my TODO list.

**Comment by junaruga** [comment]
> however I want to make sure I have automated testing of the generated gem to ensure that we're not omitting necessary files.

@flavorjones thank you for the working with this. How was the status?
I proposed it with another pull-request.

**Comment by stevecheckoway** [comment]
Making gems minimal in size sounds good to me as long as not too much is removed. E.g., https://github.com/sparklemotion/nokogiri/pull/1788 wants the headers installed in the extension directory or at least not removed.

Until that (or something like it) happens, Nokogumbo is actually using the `dependency.yml` file. See https://github.com/rubys/nokogumbo/blob/master/ext/nokogumbo/extconf.rb#L15. I'd like to remove most of that code (i.e., downloading, configuring, and installing headers for libxml2), but unless Nokogiri vends the embedded headers, having the version, and SHA-256 hash of the library source code is very helpful.

**Comment by stevecrozz** [comment]
I've rebased this branch onto most recent master and @stevecheckoway I have restored dependencies.yml to Manifest.txt as it sounds like gems may depend on it.

File: Manifest.txt [hunk 1]
```diff
@@ -256,6 +238,7 @@ patches/libxml2/0003-Fix-infinite-loop-in-LZMA-decompression.patch
 patches/sort-patches-by-date
 suppressions/README.txt
 suppressions/nokogiri_ruby-2.supp
+<<<<<<< HEAD
```
Comment (junaruga) [inline Manifest.txt]: We can remove `git` conflict mark here.

File: Manifest.txt [hunk 2]
```diff
@@ -368,3 +351,5 @@ test/xml/test_xinclude.rb
 test/xml/test_xpath.rb
 test/xslt/test_custom_functions.rb
 test/xslt/test_exception_handling.rb
+=======
+>>>>>>> Make builds minimal
```
Comment (junaruga) [inline Manifest.txt]: We can remove `git` conflict mark here.

**Comment by stevecrozz** [comment]
Thanks @junaruga. Updated.

**Comment by flavorjones** [comment]
@stevecrozz Can you please rebase on master? We fixed some things in 464717b related to JRuby 9.2 backwards incompatibilities that are making this PR fail tests on JRuby 9.2.

**Comment by stevecrozz** [comment]
No prob @flavorjones. I force-pushed a newly rebased branch.

**Comment by flavorjones** [comment]
Woo, that went green. OK, I may try to get this into v1.9.0. Thanks!

**Comment by flavorjones** [comment]
I've merged this manually! Will be in v1.9.0. Thanks again for your patience and for submitting this PR!

**Pull request closed without merging**